lib/protobuf.ex

```diff
@@ -175,7 +175,11 @@ defmodule Protobuf do
       #=> ** (Protobuf.DecodeError) ...
 
   """
-  @spec decode(binary(), message) :: %{required(:__struct__) => message} when message: module()
+  @spec decode(binary(), message) :: %{
+          required(:__struct__) => message,
+          optional(atom()) => any()
+        }
+        when message: module()
```

Comment:
I think we can change this to

```suggestion
  @spec decode(binary(), message) :: struct()
```

?

Reply:
I guess that way we'd lose the fact that the struct is not any struct but one of the `message` type.
Will fix our problem, so we can go with that if you prefer

Reply:
Aaaah, I missed that, sorry. Okay no, your fix is great. 🎉